server/entity/area_effect_cloud.go

```diff
@@ -0,0 +1,240 @@
+package entity
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity/effect"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item/potion"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"time"
+)
+
+// AreaEffectCloud is the cloud that is created when: lingering potions are thrown; creepers with potion effects explode;
+// dragon fireballs hit the ground.
+type AreaEffectCloud struct {
+	duration           int64
+	reapplicationDelay int64
+	durationOnUse      int64
+
+	radius       float64
+	radiusOnUse  float64
+	radiusGrowth float64
+
+	close bool
+
+	targets map[world.Entity]int64
+
+	t   potion.Potion
+	age int64
+
+	transform
+}
+
+// NewDefaultAreaEffectCloud ...
+func NewDefaultAreaEffectCloud(pos mgl64.Vec3, t potion.Potion) *AreaEffectCloud {
+	return NewAreaEffectCloud(pos, t, time.Second*30, time.Second*2, 0, 3.0, -0.5, -0.005)
+}
+
+// NewAreaEffectCloud ...
+func NewAreaEffectCloud(pos mgl64.Vec3, t potion.Potion, duration, reapplicationDelay, durationOnUse time.Duration, radius, radiusOnUse, radiusGrowth float64) *AreaEffectCloud {
```

Comment:
I would name `NewDefaultAreaEffectCloud` to `NewAreaEffectCloud` and `NewAreaEffectCloud` to something else, like `NewAreaEffectCloudWith` or something.

Reply:
Renamed.